assets/js-focal-image.js

```diff
@@ -9,12 +10,20 @@
  * CSS object-position uses percentages where:
  * - X: 0% (far left) to 100% (far right) with 50% being center
  * - Y: 0% (top) to 100% (bottom) with 50% being center
+ *
+ * Performance optimizations:
+ * - Uses IntersectionObserver to only process images as they enter viewport
+ * - Uses batchDOM to batch style changes
+ * - Caches computed values for efficiency
  */
 const handleFocalImages = () => {
   const focalImages = document.querySelectorAll('.focal-image');
 
   if (!focalImages || !focalImages.length) return;
 
+  // Map to cache computed values
+  const computedPositions = new Map();
```

Comment:
what about a name `positionsMapCache`? Then it's clear below that is used for caching?